libmachine/provision/ubuntu_upstart.go

```diff
@@ -135,14 +135,30 @@ func (provisioner *UbuntuProvisioner) dockerDaemonResponding() bool {
 	return true
 }
 
+func (provisioner *UbuntuProvisioner) setStorageDriver() error {
```

Comment:
Instead of duplicating this function for the two provisioners, why not make a helper function that you invoke from `Provision` method instead?  e.g. the invocation would look something like (in `Provision` method):

```golang
storageDriver, err := decideStorageDriver(provisioner, provisioner.EngineOptions.StorageDriver)
if err != nil {
    return err
}
```